proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -543,6 +545,17 @@ message Callback {
     }
 }
 
+message HSMCallbackArg {
+    // namespace ID of the workflow that just completed.
+    string namespace_id = 1;
+    // ID of the workflow that just completed.
+    string workflow_id = 2;
+    // run ID of the workflow that just completed.
+    string run_id = 3;
+    // Last event of the completed workflow.
+    temporal.api.history.v1.HistoryEvent last_event = 4;
+}
```

Comment:
This is not persisted right. Probably define it in hsm.proto.

Reply:
Also I wonder if it make sense to call this `HSMInvocationArg` and use `oneof` inside. The four existing fields can be a separate RecordHSMCompletionArg message and used inside the `oneof`. Because it feels like `hsmInvocation` is a general implementation not specific to completion callback.

Reply:
I am confused, maybe I missed something here.
I see that `HSMCallbackArg` definition is used in callback component implementation, which should be able to support calling arbitrary methods and input. 
Also if the definition is specific to completion, the name should reflect it. 
It's probably easier if we talk directly tomorrow. 

> The general HSM invocation takes arbitrary bytes

This is the remote method implementation which is on the handler side (of the callback) right?

> Probably define it in hsm.proto.

Please ignore, I forgot that hsm.proto is under persistence as well.

Reply:
Synced offline. let's rename the message so from name it's clear that it's used when HSM has completed.